client/rpc/src/eth.rs

```diff
@@ -1197,34 +1201,48 @@ impl<B, C> Web3ApiT for Web3Api<B, C> where
 	}
 }
 
-pub struct EthFilterApi<B, C> {
+pub struct EthFilterApi<B: BlockT, C, BE> {
 	client: Arc<C>,
 	filter_pool: FilterPool,
 	max_stored_filters: usize,
-	_marker: PhantomData<B>,
+	overrides: Arc<BTreeMap<EthereumStorageSchema, Box<dyn StorageOverride<B> + Send + Sync>>>,
```

Comment:
Can you move the override and fallback onto a standalone struct, and put them under a single `Arc`?